kollect-core/src/main/kotlin/kollect/Cache.kt

```diff
@@ -29,23 +29,19 @@ interface DataSourceCache {
  */
 data class InMemoryCache(val state: Map<DataSourceIdentity, Any?>) : DataSourceCache {
 
+    constructor(vararg a: Tuple2<DataSourceIdentity, Any?>): this(toMap(*a))
+
     override fun <A> get(k: DataSourceIdentity): Option<A> =
-            Option.fromNullable(state[k] as A?)
+        Option.fromNullable(state[k] as A?)
 
     override fun <A> update(k: DataSourceIdentity, v: A): DataSourceCache =
         copy(state = state.plus(Pair(k, v)))
 
     companion object {
         fun empty(): InMemoryCache = InMemoryCache(emptyMap())
 
-        fun apply(vararg results: Tuple2<DataSourceIdentity, Any?>): InMemoryCache =
```

Comment:
@fedefernandez this is called `invoke. To make it available as a special operator you also need to mark it with `operator`.

```
companion object {
     operator fun invoke(vararg results: Tuple2<DataSourceIdentity, Any?>): InMemoryCache
}
```

Reply:
Oh nice, I'll give it a try

PS: Thanks!